src/custom-lists/components/overview/sidebar/list-item.tsx

```diff
@@ -0,0 +1,169 @@
+import React, { Component, DragEventHandler } from 'react'
+import cx from 'classnames'
+
+const styles = require('./list-item.css')
+
+interface Props {
+    listName: string
+    isFiltered: boolean
+    onEditButtonClick: React.MouseEventHandler<HTMLButtonElement>
+    onCrossButtonClick: React.MouseEventHandler<HTMLButtonElement>
+    onAddPageToList: (url: string) => void
+    onListItemClick: () => void
+}
+
+interface State {
+    isMouseInside: boolean
+    isDragInside: boolean
+}
+
+class PageList extends Component<Props, State> {
+    private listItemRef: HTMLElement
+
+    constructor(props) {
+        super(props)
+        this.state = {
+            isMouseInside: false,
+            isDragInside: false,
+        }
+    }
+
+    componentDidMount() {
+        this.attachEventListeners()
+    }
+
+    componentWillUnmount() {
+        this.removeEventListeners()
+    }
+
+    private attachEventListeners() {
+        this.listItemRef.addEventListener('mouseenter', this.handleMouseEnter)
```

Comment:
Is there any reason here (and a few other places) that the ref to the real DOM element is being used to attach event listeners, rather than using the React's `onMouseEnter`/`onMouseLeave`/etc props on the virtual DOM component?

Reply:
React's `onMouseEnter/onMouseLeave` has no effect may because of shadow dom? When I first tried to re-use existing `list-item.jsx` in sidebar with `onMouseEnter/onMouseLeave` edit/share buttons on mouse enter won't show up so I attached event listeners to the `ref`

Reply:
Thanks for the explanation! Yeah, again another good place to explain the choice in a comment.